solution

miamarichka opened the pull request:
[DEMO LINK](https://miamarichka.github.io/react_movies-list-fetch-movies/)

https://mate-parttime.slack.com/archives/C041JA5EJKV/p1677323572649219 
I disagree with last suggestions, follow link for more details

mykhalenych commented on src/App.tsx:
```diff
@@ -5,7 +5,21 @@ import { FindMovie } from './components/FindMovie';
+
+    if (!isExistMovie) {
+      setIsMovieNew(true);
+      setMovies([...movies, movie]);
```
it is bad practice to use setter and getter in one place, use instead
`setMovies((prev) => [...prev, movie]);`

mykhalenych commented on src/components/FindMovie/FindMovie.tsx:
```diff
@@ -1,10 +1,100 @@
+}) => {
+  const [query, setQuery] = useState('');
+  const [movie, setMovie] = useState<Movie | null>(null);
+  const [onLoading, setOnLoading] = useState(false);
```
```suggestion
  const [isLoading, setIsLoading] = useState(false);
```

Volodymyr-Melnyk-Fron10der commented on src/App.tsx:
```diff
@@ -5,7 +5,21 @@ import { FindMovie } from './components/FindMovie';
+  const [movies, setMovies] = useState<Movie[]>([]);
+  const [isMovieNew, setIsMovieNew] = useState(true);
+
+  const addMovie = (movie: Movie) => {
```
keep best practice naming
```suggestion
  const handleAddMovie = (movie: Movie) => {
```

Volodymyr-Melnyk-Fron10der replied:
check it throughout your code

miamarichka replied:
We don`t handle  something specific which occurs on and up to action, I pass it as props, I don`t call it in current component

Volodymyr-Melnyk-Fron10der commented on src/App.tsx:
```diff
@@ -5,7 +5,21 @@ import { FindMovie } from './components/FindMovie';
+
+  const addMovie = (movie: Movie) => {
+    const isExistMovie = movies.find(
+      existMovie => existMovie.imdbId === movie.imdbId,
```
use decomposing structure

miamarichka replied:
I rather use without decomposing in this case because its make my code more readibly and understantable

Volodymyr-Melnyk-Fron10der commented on src/App.tsx:
```diff
@@ -5,7 +5,21 @@ import { FindMovie } from './components/FindMovie';
+    const isExistMovie = movies.find(
+      existMovie => existMovie.imdbId === movie.imdbId,
+    );
+
+    if (!isExistMovie) {
+      setIsMovieNew(true);
+      setMovies((prevMovies) => [...prevMovies, movie]);
+    } else {
+      setIsMovieNew(false);
+    }
```
simplify condition
```suggestion
      setIsMovieNew(!isExistMovie);
      setMovies((prevMovies) => [...prevMovies, movie]);
```

miamarichka replied:
Would like to shorter the way the show but unfortunately it can break the code, because short in this way leads lack of some settings as well

Volodymyr-Melnyk-Fron10der replied:
```
  setIsMovieNew(!isExistMovie);
  if (!isExistMovie) {
      setMovies((prevMovies) => [...prevMovies, movie]);
  }
```

Volodymyr-Melnyk-Fron10der commented on src/components/FindMovie/FindMovie.tsx:
```diff
@@ -1,10 +1,100 @@
+import { Movie } from '../../types/Movie';
+
+type Props = {
+  onAddMovieBtn: (newMovie: Movie) => void;
```
```suggestion
  onAddMovie: (newMovie: Movie) => void;
```

miamarichka replied:
I use 2 function addMovie, and better to show which one is going to perform actions causes directly on button to make it understanble

Volodymyr-Melnyk-Fron10der commented on src/components/FindMovie/FindMovie.tsx:
```diff
@@ -1,10 +1,100 @@
+
+type Props = {
+  onAddMovieBtn: (newMovie: Movie) => void;
+  thisMovieExist: boolean
```
it's better to pass callback like `checkMovieExist` which will check is movie in your movies and return boolean value

miamarichka replied:
Instead of one short state word you offer me to create a separete function which one return a boolean, it dosent seems needed

Volodymyr-Melnyk-Fron10der commented on src/components/FindMovie/FindMovie.tsx:
```diff
@@ -1,10 +1,100 @@
+  const [query, setQuery] = useState('');
+  const [movie, setMovie] = useState<Movie | null>(null);
+  const [isLoading, setIsLoading] = useState(false);
+  const [onError, setOnError] = useState(false);
```
```suggestion
  const [isError, setIsError] = useState(false);
```

Volodymyr-Melnyk-Fron10der commented on src/components/FindMovie/FindMovie.tsx:
```diff
@@ -1,10 +1,100 @@
+  const [isLoading, setIsLoading] = useState(false);
+  const [onError, setOnError] = useState(false);
+
+  const createNewMovie = (foundMovieData:MovieData | null) => {
```
`handle...`

Volodymyr-Melnyk-Fron10der commented on src/components/FindMovie/FindMovie.tsx:
```diff
@@ -1,10 +1,100 @@
+    }
+  };
+
+  const loadMovie = async (searchQuery:string) => {
```
handle

Volodymyr-Melnyk-Fron10der commented on src/components/FindMovie/FindMovie.tsx:
```diff
@@ -1,10 +1,100 @@
+  };
+
+  const onInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    if (onError && query === e.target.value) {
+      setOnError(true);
+    } else {
+      setOnError(false);
+    }
```
```suggestion
      setOnError(onError && query === e.target.value);
```

miamarichka replied:
I need to set it fasle, not only check

Anastasiia-Svintsova replied:
So `onError && query === e.target.value` equal to true or false. 
`setOnError(onError && query === e.target.value);` will work

Volodymyr-Melnyk-Fron10der commented on src/components/FindMovie/FindMovie.tsx:
```diff
@@ -1,10 +1,100 @@
+    loadMovie(query);
+  };
+
+  const onInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
```
```suggestion
  const onInputChange = ({ target: { value } }: React.ChangeEvent<HTMLInputElement>) => {
```

miamarichka replied:
I rather use without decomposing in this case because its make my code more readibly and understantable

Volodymyr-Melnyk-Fron10der commented on src/components/FindMovie/FindMovie.tsx:
```diff
@@ -1,10 +1,100 @@
+  };
+
+  const addMovie = (movieToAdd: Movie) => {
+    if (thisMovieExist) {
```
```suggestion
    if (checkMovieExist(movieToAdd.id)) {
```

Volodymyr-Melnyk-Fron10der commented on src/components/FindMovie/FindMovie.tsx:
```diff
@@ -17,41 +107,54 @@ export const FindMovie: React.FC = () => {
+              disabled={query.trim() === ''}
             >
-              Find a movie
+              {!movie
```
handle truth case first

Volodymyr-Melnyk-Fron10der commented on src/utils.tsx:
```diff
@@ -0,0 +1,3 @@
+export const KEY = '8550fef4';
+export const DEFAULT_IMG
+    = 'https://via.placeholder.com/360x270.png?text=no%20preview';
```
filename: `src/constants/common.ts`

Anastasiia-Svintsova commented on src/components/FindMovie/FindMovie.tsx:
```diff
@@ -17,41 +107,54 @@ export const FindMovie: React.FC = () => {
-              className="button is-light"
+              className={classNames('is-light', 'button',
+                { 'is-loading': isLoading })}
+              disabled={query.trim() === ''}
```
One of the possible ways. Just my suggestion. 
```suggestion
              disabled={!query.trim().length}
```

Anastasiia-Svintsova reviewed (changes requested):
- I can't add any movies after I added some of them 2 times. I hope you agree that it shouldn't be like this 🙏 

https://user-images.githubusercontent.com/99403198/221357049-ff3455e2-541b-4366-8713-0d35042676b0.mov

I wrote small suggestions in the code, but it's just suggestions. The main reason to reject it is the bug described above. Keep up the good job

miamarichka commented:
> * I can't add any movies after I added some of them 2 times. I hope you agree that it shouldn't be like this 🙏

https://icecreamapps.com/v/f3m553f


Check, there is no bag . It can ups to your connection

miamarichka commented:
> > * I can't add any movies after I added some of them 2 times. I hope you agree that it shouldn't be like this 🙏
> 
> https://icecreamapps.com/v/f3m553f
> 
> Check, there is no bag . It can ups to your connection

Any why there is no adress on video, my DEMOLINK adress

miamarichka commented:
> [DEMO LINK](https://miamarichka.github.io/react_movies-list-fetch-movies/)
> 
> https://mate-parttime.slack.com/archives/C041JA5EJKV/p1677323572649219 I disagree with last suggestions, follow link for more details

Please, check your connection before ask me to fix bugs https://icecreamapps.com/v/f3m553f

miamarichka commented:
> * I can't add any movies after I added some of them 2 times. I hope you agree that it shouldn't be like this 🙏
> 
> I wrote small suggestions in the code, but it's just suggestions. The main reason to reject it is the bug described above. Keep up the good job

I fixed it

polosanya reviewed (approved):
Good job! I'm approving it, but please pay attention to the comments. 
Proper naming, code styling and formatting are also important part of clean and understandable code. If you disagree with some comments, please feel free to reply to them and re-request a review. But your response should be supported by documentation, an article or any other source, not just your opinion. I hope this way will help to improve communication and make our work more productive 🙂
